sys-boot/udk/files/udk-workspace.template

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+show_usage() {
+	if [ ! -z "${1}" ]; then
+		echo "${1}" >&2
+		echo >&2
+	fi
+	echo "Usage:"
+	echo ". ${BASH_SOURCE[0]} [folder]"
+	echo "where folder is the workspace folder - current directory is used if unspecified"
+}
+
+create_workspace() {
+	local workspace_dir="${PWD}"
+	if [ ! -z "${1}" ]; then
+		workspace_dir="${1}"
+		if [ ! -d "${1}" ]; then
+			if ! mkdir -p "${1}"; then
+				show_usage "Cannot create directory ${1}"
+				return 1
+			fi
+		fi
+	fi
+
+	pushd "${workspace_dir}" >>/dev/null
+	if [ ! -d Conf ] && ! mkdir Conf; then
+		show_usage "Cannot create configuration directory"
+		return 1
+	fi
+	export WORKSPACE="${PWD}"
+	popd >>/dev/null
+}
+
+if [ "${1}" = "-h" ] || [ "${1}" = "--help" ]; then
+	show_usage
+	# Script may have been sourced
+	return 0 2>>/dev/null
```

Comment:
Hmm, I didn't know you can `return` from sourcing. Is this portable? If you're not sure, then it would probably be better to wrap it into `main()` function, and `return` from that.

Reply:
I would not be able to give an error code on script exit…

Reply:
Why not? Just `return` from `main()` with an exit code, and the script's exit code will be the exit code of last command executed, i.e. `main`.

Reply:
Good idea, but then, I will not be able to unset main() and other functions. Should I prefix them and not unset them?

Reply:
I've checked with POSIX and `return` is valid here.

Reply:
Great, so I'll keep as is.